synapse/handlers/space_summary.py

```diff
@@ -378,8 +378,46 @@ async def _get_room_hierarchy(
 
                 processed_rooms.add(room_id)
             else:
-                # TODO Federation.
-                pass
+                # If a previous call got information for this room *and* it is
+                # not a space (or the max-depth has been achieved), include it.
+                if queue_entry.remote_room and (
+                    queue_entry.remote_room.get("room_type") != RoomTypes.SPACE
```

Comment:
Yep, I'll include it here too. The `remote_room` is:

> The room summary for this room returned via federation. This will only be
> used if the room is not known locally (and is not a space).